src/main/java/io/vertx/core/impl/WorkerExecutor.java

```diff
@@ -22,6 +24,18 @@
  */
 public class WorkerExecutor implements EventExecutor {
 
+  public static io.vertx.core.impl.WorkerExecutor unwrapWorkerExecutor() {
+    ContextInternal ctx = (ContextInternal) Vertx.currentContext();
+    if (ctx != null) {
+      ctx = ctx.unwrap();
+      Executor executor = ctx.executor();
+      if (executor instanceof io.vertx.core.impl.WorkerExecutor) {
+        return (io.vertx.core.impl.WorkerExecutor) executor;
+      }
+    }
+    throw new IllegalStateException("Not running on a Vert.x virtual thread");
```

Comment:
This methods works even for any type of `WorkerExecutor`, shouldn't the message be updated?

Reply:
yes it should fail with specific message though (event-loop vs plain thread)